classes/shop/Shop.php

```diff
@@ -135,7 +138,9 @@ public function __construct($id = null, $id_lang = null, $id_shop = null)
 
         if ($this->id) {
             $this->setUrl();
-            $this->setTheme();
+            if ($this->theme == null) {
```

Comment:
I'm pretty sure this variable always equals null.
This is a attribute of this class, and nothing can alter it before we reach this line.